src/averageAges.js

```diff
@@ -1,61 +1,73 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredAges = people.filter((person) =>
+    person.sex === 'm'
+    && (century ? Math.ceil(person.died / 100) === century : true)
+  );
+
+  return filteredAges.reduce((sum, person) =>
+    sum + person.died - person.born, 0) / filteredAges.length;
```

Comment:
the same, not fixed
<img width="756" alt="image" src="https://user-images.githubusercontent.com/102411130/230006265-8e4ce582-240a-4a6b-bc8c-04426f48b68d.png">

move this calculation to a separate helper function and use it in each function you need as you make the same calculation three times `reduce((sum, person) =>
    sum + person.died - person.born, 0) / filteredAges.length;`